app/controllers/api/v1/pins_controller.rb

```diff
@@ -16,9 +16,28 @@ def create
       end
     end
 
+    def update
+      pin = Pin.find_by_id(params[:id])
+      pin.icon = params[:icon]
+      if pin.valid?
+        pin.save!
```

Comment:
this line will produce an exception if there is invalid validation so we should use try/catch and return with status code unprocessable_entity(422) otherwise